pkg/preparer/setup.go

```diff
@@ -404,6 +404,22 @@ func New(preparerConfig *PreparerConfig, logger logging.Logger) (*Preparer, erro
 		return nil, util.Errorf("Could not create preparer pod directory: %s", err)
 	}
 
+	// Artifact files are downloaded to os.TempDir().
+	// Since we extract artifact files as target user, we must allow them to access the tmpdir.
+	// We expect that there is no sensitive information in TempDir, so 755 is safe, though 711 could be considered.
+	tmpDirStat, err := os.Stat(os.TempDir())
+	if err != nil {
+		return nil, util.Errorf("Could not stat tmpdir: %s", err)
+	}
+	mode := tmpDirStat.Mode()
+	if mode&0755 != 0755 {
+		// keep whatever upper bit is there.
+		err = os.Chmod(os.TempDir(), (mode&07000)|0755)
```

Comment:
wondering if anyone thinks this (keeping the highest... triad) is necessary. This was added after the most recent approval, would like a quick look at it.